tests/robustness/coverage/coverage_test.go

```diff
@@ -16,20 +16,111 @@ package coverage_test
 
 import (
 	"bytes"
+	"cmp"
 	"encoding/json"
 	"fmt"
+	"maps"
 	"os"
 	"path/filepath"
+	"slices"
+	"strconv"
 	"strings"
 	"testing"
 
+	gocmp "github.com/google/go-cmp/cmp"
 	"github.com/olekukonko/tablewriter"
 	"github.com/olekukonko/tablewriter/tw"
 	traceservice "go.opentelemetry.io/proto/otlp/collector/trace/v1"
 	tracev1 "go.opentelemetry.io/proto/otlp/trace/v1"
 	"google.golang.org/protobuf/encoding/protojson"
 )
 
+type refOp struct {
+	// args splits traces into buckets based on Matched results.
+	args map[string]Matched
+	// wantArgs lists the expected buckets stored in args.
+	wantArgs map[string]struct{}
+	// methods tries to associate Matched with method name.
+	methods map[string]Matched
+	// knownBypasses filters out traces from the args analysis.
+	knownBypasses map[string]Matched
+}
+
+var referenceUsageOfEtcdAPI = map[string]refOp{
+	"etcdserverpb.KV/Range": {
+		// All calls should go through etcd-k8s interface
+		args: map[string]Matched{
+			"countOnly": isCountOnly,
+			"keysOnly":  isKeysOnly,
+			"limit":     isLimitSet,
+			"limit=1":   keyIsEqualInt("limit", 1),
+			"rangeEnd":  isRangeEndSet,
+			"rev":       isRevisionSet,
+		},
+		wantArgs: map[string]struct{}{
+			"countOnly=false,keysOnly=false,limit=false,limit=1=false,rangeEnd=true,rev=false": {},
+			"countOnly=false,keysOnly=false,limit=true,limit=1=false,rangeEnd=true,rev=false":  {},
+			"countOnly=false,keysOnly=false,limit=true,limit=1=false,rangeEnd=true,rev=true":   {},
+			"countOnly=false,keysOnly=false,limit=true,limit=1=true,rangeEnd=false,rev=false":  {},
+			"countOnly=false,keysOnly=false,limit=true,limit=1=true,rangeEnd=true,rev=true":    {},
+			"countOnly=true,keysOnly=false,limit=false,limit=1=false,rangeEnd=true,rev=false":  {},
+		},
+		methods: map[string]Matched{
+			"Get": andMatcher(
+				keyIsEqualInt("limit", 1),
+				notMatcher(isRangeEndSet),
+			),
+			"Count": andMatcher(
+				isCountOnly,
+				isRangeEndSet,
+				notMatcher(orMatcher(isKeysOnly, isLimitSet, isRevisionSet)),
+			),
+			"List": andMatcher(
+				isRangeEndSet,
+				notMatcher(orMatcher(isKeysOnly, isCountOnly, keyIsEqualInt("limit", 1))),
+			),
+			"List(ConsistentRead)": andMatcher(
```

Comment:
```suggestion
			"GetCurrentRevision": andMatcher(
```

To match https://github.com/kubernetes/kubernetes/blob/8e6d788887034b799f6c2a86991a68a080bb0576/staging/src/k8s.io/apiserver/pkg/storage/interfaces.go#L269

Reply:
My initial idea was to point to the method of contract interface (and then maybe the special context when it is used)